tests/unit/test_server.py

```diff
@@ -40,28 +40,121 @@ class TestInstall(BaseUnitCase):
 
     def setUp(self):
         self.remove_runs_once_flag(server.status)
+        self.remove_runs_once_flag(server.install)
         self.maxDiff = None
         super(TestInstall, self).setUp(capture_output=True)
 
+    @patch('prestoadmin.server._find_local_presto_rpm')
     @patch('prestoadmin.server.package.check_if_valid_rpm')
     @patch('prestoadmin.server.execute')
-    def test_install_server(self, mock_execute, mock_check_rpm):
-        local_path = "/any/path/rpm"
-        server.install(local_path)
-        mock_check_rpm.assert_called_with(local_path)
+    def test_install_server_local_path(self, mock_execute, mock_check_rpm,
+                                       mock_find_local_rpm):
+        rpm_specifier = "/any/path/rpm"
+        mock_find_local_rpm.return_value = rpm_specifier
+        env.rpm_source = 'facebook'
+        server.install(rpm_specifier)
+        mock_check_rpm.assert_called_with(rpm_specifier)
         mock_execute.assert_called_with(server.deploy_install_configure,
-                                        local_path, hosts=get_host_list())
+                                        rpm_specifier, hosts=get_host_list())
+
+    @patch('prestoadmin.server._find_local_presto_rpm')
```

Comment:
My rationale for creating all of the mock objects was that the rpm_specifier can take on a few different forms and I wanted to use the mocks to control which form was used for the test. For instance, if I made rpm_specifier = 0.148, I want it to be interpreted as a version, so I make sure that finding it locally fails. If I want it to check the download URL, I make sure that it doesn't find a local downloaded copy. In both these cases, I have to mock those functions so I can set their return values to signal that they failed. This way, I make sure I'm testing whatever single function I'm trying to test. Isn't that what I want from a unit test?

What you're suggesting is mocking the lower-level functions in the higher-level ones. If we wanted to do that, wouldn't it be better to have these tests mock the higher level functions like `find_local_presto_rpm` and `check_if_valid_rpm`, and then create separate unit tests for these higher level functions where we mock `os.path.isfile` and `package.local`?